src/collector.rs

```diff
@@ -0,0 +1,85 @@
+extern crate math;
+
+use std::{
+    cell::{Cell, UnsafeCell},
+    collections::HashMap
+};
+
+use std::{
+    alloc::{alloc, Layout},
+    path::Path
+};
+
+use crate::{
+    safepoints::{ReturnAddress, SafepointRoots},
+    Scan
+};
+
+/// The size of the heap in bytes
+const HSIZE: usize = 1024;
+
+/// The byte alignment of the heap
+const HALIGN: usize = 4;
+
+pub(crate) struct Collector {
+    hptr: UnsafeCell<*mut usize>,
```

Comment:
Can we not make this `Cell` and use `set` and so on?

Reply:
Yes, I'm not sure why I left this one as `UnsafeCell` when changing the others to `Cell`. I've pushed a new commit to address this now.